src/spectral.jl

```diff
@@ -264,3 +264,22 @@ function contract(nbt::Nonbacktracking, edgespace::Vector)
     contract!(y,nbt,edgespace)
     return y
 end
+
+"""spectral_distance(G₁, G₂ [, k])
+Compute the spectral distance between graphs G₁ and G₂
+using the top k greatest eigenvalues. If k is ommitted,
+uses k = min(|G₁|,|G₂|).
+
+For further details, please refer to:
+
+JOVANOVIC, I.; STANIC, Z., 2014. Spectral Distances of
+Graphs Based on their Different Matrix Representations
+"""
+function spectral_distance(G₁::SimpleGraph, G₂::SimpleGraph, k::Integer)
+  λ₁ = adjacency_spectrum(G₁)[end-k+1:end]
```

Comment:
This is a dense matrix operation see the warning at https://github.com/JuliaGraphs/LightGraphs.jl/blob/master/src/spectral.jl#L79
Since you only want the top k eigenvectors, maybe you should use `lambda1 = eigs(adjacency_matrix(g), nev=k, ncv = maximum(20,2*k))[1]` for both sets of eigenvalues.